aws/resource_aws_iam_user.go

```diff
@@ -174,6 +181,31 @@ func resourceAwsIamUserUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("tags") {
+		// Reset all tags to empty set
+		oraw, nraw := d.GetChange("tags")
+		o := oraw.(map[string]interface{})
+		n := nraw.(map[string]interface{})
+		c, r := diffTagsIAM(tagsFromMapIAM(o), tagsFromMapIAM(n))
+
+		_, untagErr := iamconn.UntagUser(&iam.UntagUserInput{
```

Comment:
We should perform length checks here to ensure we don't try to call `UntagUser` with an empty set of `TagKeys`, e.g.

```go
if len(r) > 0 {
  // existing logic
}
```